prompt.go

```diff
@@ -270,23 +270,27 @@ func (p *Prompt) feed(buf []byte) (shouldExit bool, exec *Exec, bufLeft []byte,
 		if p.handleASCIICodeBinding(b) {
 			return
 		}
-		i := 0
-		r, _ := utf8.DecodeRune(b[i:])
-		// AltGr key handling: expect ESC (0x1b) followed by Unicode character; skip ESC
-		if r == 0x1b {
-			i = 1
-			r, _ = utf8.DecodeRune(b[i:])
-		}
+		r, _ := nextRuneSkippingLeadingEsc(b)
 		// Only insert printable characters
-		if strconv.IsPrint(r) {
-			p.buf.InsertText(string(b[i:]), false, true)
+		if r != utf8.RuneError && strconv.IsPrint(r) {
+			p.buf.InsertText(string(r), false, true)
```

Comment:
One thing worth noting:

There was some subtle behavior here that was made evident by Inserting `r` instead of `b`.

In some cases, like with line termination, `DecodeRune()` would return `RuneError`. And unfortunately `IsPrint()` returns `true` for `RuneError`. In the old code this didn't cause a problem because `string(RuneError)` returned an empty string, but with the change of inserting `r` instead of (a slice of) `b`, `string(r)` returns an unprintable placeholder character. This is now avoided with the additional check that `r != RuneError`.

Reply:
Ok, yeah, that's surprising (and disappointing) about `RuneError` returning true for `IsPrint`.